### ISSUE TITLE
Change marker colors

### ISSUE THREAD

**Issue by alexgleason** [Rudloff/openvegemap#40]
Hi, thanks for creating this! A vegan activist group I'm part of just sat down and mapped out all the vegan places in Philadelphia two days ago, so this came at just the right time. :smile:

![screenshot from 2018-05-07 11 19 32](https://user-images.githubusercontent.com/3639540/39710166-0413671a-51ea-11e8-8743-c873f900094f.png)

There are a lot of all-vegan places in Philly and we want the ability to distinguish them from the rest. I'm a big fan of how [vegphilly](http://www.vegphilly.com/vendors/) handles this:

![screenshot from 2018-05-07 11 23 09](https://user-images.githubusercontent.com/3639540/39709869-11ddc3aa-51e9-11e8-946d-7b7c499321c4.png)

* Green is `diet:vegan=only`. As a vegan, this is my first choice for where I want to go. I can eat everything here.
* Orange is `diet:vegetarian=only` && `diet:vegan=yes`. I hate to be around dead animals, and this place will likely have a lot of options for me. It's all-vegetarian with good vegan options, so this is my second choice.
* Yellow is `diet:vegan=yes`. I would only go here if I'm desperate. This place serves dead animals and I might have to barter with the waiters to make them understand my needs. I only care about going here if I really must, so it should be a different color.

**Comment by Rudloff** [comment]
Hello,

Thanks for your proposal but I think a lot of people use this app to look for a veg-friendly option and don't mind if they serve meat. So I would rather stick with the vegan/vegetarian/meat colors which are easy to understand.
(I also find vegphilly's legend a bit confusing. When I first looked at the website I wondered "why are places without vegetarian food listed on a website like this?".)

Also, if I understand correctly, vegphilly.com does not display places that serve vegetarian food but no vegan food, right? While I understand this decision, OpenVegeMap is an app for vegans and for vegetarians.

OpenVegeMap provides filters that you can use to only display places that don't serve meat (if you only enable *Vegan only* and *Vegetarian only*). Wouldn't this be enough for your needs?

**Comment by alexgleason** [comment]
I'm thinking we might disagree because of regional differences.

Philadelphia has **20+ all vegan restaurants**. There are so many vegan restaurants that it doesn't make sense to go somewhere that isn't 100% vegan.

>I think a lot of people use this app to look for a veg-friendly option and don't mind if they serve meat.

In densely populated US cities like Philadelphia, New York City, Boston, and San Francisco, people really just want to visit all-vegan restaurants. We don't want to haggle with the wait-staff "is this vegan? Can you please make this vegan? Are you *sure* you prepared this vegan? Can you check the ingredients?" It's so annoying to haggle so they don't kill and abuse animals.

>OpenVegeMap provides filters that you can use to only display places that don't serve meat

I didn't know about this, thanks! Still, I think the default experience is important.

>I also find vegphilly's legend a bit confusing.

It is actually very similar to the legend used by HappyCow. Here's Philly on HappyCow:

![screenshot from 2018-05-07 14 17 36](https://user-images.githubusercontent.com/3639540/39717837-6ef630b8-5202-11e8-80b6-4fc275825178.png)

* Green is 100% vegan
* Purple is 100% vegetarian
* Red is vegan/vegetarian options

HappyCow is the most popular website for this, so I think there are some user expectations already in place about how a veg map should look.

>Also, if I understand correctly, vegphilly.com does not display places that serve vegetarian food but no vegan food, right?

Actually this is vegphilly's colors:

* Green is 100% vegan
* Orange is 100% vegetarian
* Yellow is vegan/vegetarian options

As you can see it's nearly the same as HappyCow. I think yellow is a bit forgiving than red for veg options.

>OpenVegeMap is an app for vegans and for vegetarians.

Okay, my previous suggestion had a stronger preference for veganism, because I thought this was mostly for vegans. Who are the users of this app? Vegans and vegetarians will prefer HappyCow because it has a lot more features, more complete data, and is more widely known. The only people who will use this are free software enthusiasts like myself who care about freedom and privacy. However, if this app makes the experience a lot better for vegans specifically, vegans will prefer this app over HappyCow even if they don't care about software freedom. Personally I think it makes sense to focus on veganism because you could fill this much-needed niche.

Still, if you don't want to do that, I think switching to a color scheme more similar to HappyCow or VegPhilly would improve it a lot since it would line up more with user expectations.

So in conclusion: HappyCow and VegPhilly display vegan options as red and yellow respectively, while OpenVegeMap displays them as green. In my opinion, users want 100% vegan places to be green, and veg options to be a different color.

Thanks again for your work on this. :smile: It's okay if we still disagree, I'm just glad there's a FLOSS solution in the works!

**Comment by Rudloff** [comment]
> There are so many vegan restaurants that it doesn't make sense to go somewhere that isn't 100% vegan.

You can already spot vegan-only places on the map (light green + dot-in-a-circle icon).

>In my opinion, users want 100% vegan places to be green, and veg options to be a different color.

We already have different greens for vegan and vegetarian.
(Also red is already used by our `vegetarian=no` filter.)

Sorry but I don't really see the benefit of changing the color scheme (apart from matching what some other websites use). Is there some info that you can't get with the current map?

**Comment by alexgleason** [comment]
>Also red is already used by our vegetarian=no filter.

I was wondering about this too, why would a marker with `diet:vegetarian=no` ever show up on the map? Any example?

>Is there some info that you can't get with the current map?

No, I can definitely get the info I need, my issue is about the presentation of that info. Even if vegan-only restaurants have a gold star instead of a green bullseye this would improve it IMO.

**Comment by Rudloff** [comment]
> I was wondering about this too, why would a marker with diet:vegetarian=no ever show up on the map? Any example?

It is mostly used by contributors in order to see if a place has already been checked (the filter is disabled by default).

> No, I can definitely get the info I need, my issue is about the presentation of that info. Even if vegan-only restaurants have a gold star instead of a green bullseye this would improve it IMO.

I agree the current icons are not very meaningful. They were added to help colorblind people (https://github.com/Rudloff/openvegemap/issues/21). I am certainly open to suggestions on how to improve them (as long as they remain accessible).

**Comment by Rudloff** [comment]
I wanted to use different leaf/vegetable icons like HappyCow does but we use [FontAwesome](https://fontawesome.com/icons) and it has very few food/plant icons.

**Comment by Rudloff** [comment]
I still think this is too opinionated towards veganism (orange kind of sends a mixed message to vegetarians and you don't display vegetarian-but-not-vegan places).
I totally understand where you're coming from but like I said, I want OpenVegeMap to be for vegans and vegetarians.

I you feel so strongly about this, maybe the fork is indeed your best option.
Please tell me if you I can make maintaining a fork easier. (I want the app to be easy to fork, so it can forked by other communities like gluten-free of kosher.)

**Comment by Rudloff** [comment]
I'm closing this since you now have your fork. But feel free to open new issues if you have other suggestions.
